OsmAnd/src/net/osmand/plus/OsmandApplication.java

```diff
@@ -1060,23 +1060,31 @@ public String getUserAndroidId() {
 		return userAndroidId;
 	}
 
-	public void logEvent(String event) {
+	public void logEvent(@NonNull String event) {
 		try {
 			analyticsHelper.addEvent(event, AnalyticsHelper.EVENT_TYPE_APP_USAGE);
 		} catch (Exception e) {
 			LOG.error(e);
 		}
 	}
 
-	public void logMapDownloadEvent(String event, IndexItem item) {
+	public void logRoutingEvent(@NonNull String event) {
+		try {
+			analyticsHelper.addEvent(event, AnalyticsHelper.EVENT_TYPE_ROUTING);
```

Comment:
Is here timestamp added?

Reply:
Yes, it is added in AnalyticsHelper.addEvent() - https://github.com/osmandapp/Osmand/blob/master/OsmAnd/src/net/osmand/plus/helpers/AnalyticsHelper.java#L266